File: Angular/app-252/src/app/models/person.ts
export class Person {
    constructor(
        public firstName: string,
        public surName: String,
        public dni: string,
        public birthDate: string,
        public email: string,
        public status: boolean,
        public phone: string
    ){}
    esFullName(): string {
        return `${this.firstName} ${this.surName}`;
    }
}

File: mi-proyecto/mi-app/src/main/java/com/epiis/mi_app/model/Person.java
package com.epiis.mi_app.model;

import java.io.Serializable;
import java.time.LocalDate;

import jakarta.persistence.Column;

import jakarta.persistence.Id;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.Setter;

@MappedSuperclass
@Getter
@Setter
public abstract class Person implements Serializable {
    private static final long serialVersionUID = 1L;
    @Id
    @Column(name = "id_person", nullable = false, unique = true, length = 50)
    private String idPerson;
    @Column(name = "firstname", nullable = false)
    private String firstName;
    @Column(name = "surname", nullable = false)
    private String surname;
    @Column(name = "dni", length = 8, nullable = false)
    private String dni;
    @Column(name = "phone", nullable = false)
    private String phone;
    @Column(name = "birth_date", nullable = false)
    private LocalDate birthDate;

}
